### ISSUE TITLE
train_sklearn_model.py

### ISSUE THREAD

**Issue by Nikitabhatkudav** [Nikitabhatkudav/Stueden_Performance_Prediction_System_DSLC#1]
import os
import pandas as pd
import numpy as np
from sklearn.model_selection import train_test_split
from sklearn.preprocessing import StandardScaler
from sklearn.neural_network import MLPClassifier
from sklearn.metrics import accuracy_score, classification_report
import joblib

DATA_PATH = os.path.join('data', 'students_enhanced.csv')
MODEL_PATH = 'student_model.pkl'
SCALER_PATH = 'scaler.pkl'

def load_data(path: str):
    """Load and preprocess the enhanced dataset"""
    df = pd.read_csv(path)
    X = df[['study_hours', 'attendance', 'total_marks', 'obtained_marks', 'subject_difficulty']].astype(float).values
    y = df['pass'].astype(int).values
    return X, y

def build_model():
    """Build MLPClassifier (neural network) with scikit-learn"""
    model = MLPClassifier(
        hidden_layer_sizes=(16, 8, 4),
        activation='relu',
        solver='adam',
        alpha=0.01,
        batch_size=8,
        learning_rate='adaptive',
        max_iter=500,
        random_state=42
    )
    return model

def train_and_save():
    """Train the enhanced model and save artifacts"""
    print("Loading enhanced dataset...")
    X, y = load_data(DATA_PATH)
    
    print(f"Dataset shape: {X.shape}, Target distribution: {np.bincount(y)}")
    
    # Scale features
    scaler = StandardScaler()
    X_scaled = scaler.fit_transform(X)
    
    # Split data
    X_train, X_val, y_train, y_val = train_test_split(
        X_scaled, y, test_size=0.2, random_state=42, stratify=y
    )
    
    print("Building enhanced neural network model...")
    model = build_model()
    
    print("Training model...")
    model.fit(X_train, y_train)
    
    # Evaluate model
    y_val_pred = model.predict(X_val)
    acc = accuracy_score(y_val, y_val_pred)
    print(f'\nValidation Accuracy: {acc:.3f}')
    print('\nClassification Report:')
    print(classification_report(y_val, y_val_pred, target_names=['Fail', 'Pass']))
    
    # Save model and scaler
    joblib.dump(model, MODEL_PATH)
    joblib.dump(scaler, SCALER_PATH)
    print(f'\nSaved model to {MODEL_PATH}')
    print(f'Saved scaler to {SCALER_PATH}')

if __name__ == '__main__':
    os.makedirs('data', exist_ok=True)
    train_and_save()
